### PR TITLE
GitHub Actions workflow improvements

### PR DESCRIPTION
**Pull request by jgarber623**
This PR refactors the existing CI workflow. Notable changes include:

- renaming the file to `ci.yml` to better reflect its function,
- replaces much of the setup-related job steps with [CargoSense/setup-elixir-project](https://github.com/CargoSense/setup-elixir-project) reusable Action,
- adds `workflow_dispatch` as an event trigger so that we may manually invoke the Action (should that prove useful)

~~I've retained the use of a matrix even though there's only one configured OTP and Elixir version. @billylanchantin and I have been chatting about what an improved test matrix might look like, but haven't yet come to a conclusion. One we do, I'm happy to update the matrix with additional versions.~~

Updated in c9e6068. Pulling out my commit message since it includes some relevant notes:

> This required using an older version of Ubuntu to support older Erlang/OTP versions. Pulled that bit of knowledge from the README here:
>
> https://github.com/erlef/setup-beam
> 
> Additionally, Erlang/OTP compatibility charts are maintained here:
> 
> https://hexdocs.pm/elixir/compatibility-and-deprecations.html#between-elixir-and-erlang-otp
> 
> …and lastly, the matrix expansion via `include` is documented here:
> 
> https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#expanding-or-adding-matrix-configurations
> 
> I'm _mostly_ confident this will work as desired.

### CODE REVIEW

**Comment by jgarber623** [comment]
I suspect the changes to this workflow aren't running as part of the PR status checks because I renamed the file. 🤦🏻

File: .github/workflows/elixir.yaml [hunk 1]
```diff
@@ -1,85 +0,0 @@
...
-          cache-name: cache-elixir-deps
-        with:
-          path: deps
-          key: ${{ runner.os }}-mix-${{ env.cache-name }}-${{ hashFiles('**/mix.lock') }}
-          restore-keys: |
-            ${{ runner.os }}-mix-${{ env.cache-name }}-
-
-      # Step: Define how to cache the `_build` directory. After the first run,
-      # this speeds up tests runs a lot. This includes not re-compiling our
-      # project's downloaded deps every run.
```
Comment (billylanchantin) [inline .github/workflows/elixir.yaml:54]: How come we've removed the `_build` caching?

Comment (jgarber623) [reply .github/workflows/elixir.yaml:54]: Caching the `_build` folder should be handled by the setup-elixir-project action:

https://github.com/CargoSense/setup-elixir-project/blob/main/action.yml#L71-L77

Comment (jgarber623) [reply .github/workflows/elixir.yaml:54]: If it doesn't, that's a bug to be sorted out.

Comment (billylanchantin) [reply .github/workflows/elixir.yaml:54]: OH! I misunderstood how we were using `setup-elixir-project`. Ignore me :)

Comment (jgarber623) [reply .github/workflows/elixir.yaml:54]: It's okay! The reusable action does _a lot_ so I had to look that up to verify.

**Review by billylanchantin — APPROVED** [review]
Looks good to me!

If we can't get the CI to run on this branch, I'm happy to merge into `main` then test later. We can always revert if it doesn't do what we want.

**Pull request merged**